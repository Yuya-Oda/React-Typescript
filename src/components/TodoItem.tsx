import React from 'react';
import { Todo } from '../model/Todo';
import styles from './TodoItem.css';

type Props = {
  todo: Todo;
  onclick?: () => void;
};

export const TodoItem: React.FC<Props> = ({ todo, onclick }) => {
  return (
    <div className={styles.todoItem} onClick={onclick}>
      <div className={styles.todoItemTitle}>{todo.title}</div>
      <div className={styles.todoDescriptionInput}>{todo.description}</div>
    </div>
  );
};
