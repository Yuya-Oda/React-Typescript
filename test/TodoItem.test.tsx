import { TodoItem } from '../src/components/TodoItem';
import { Todo } from '../src/model/Todo';
import '@testing-library/jest-dom';
import { screen, render, cleanup } from '@testing-library/react';
import React from 'react';
import userEvent from '@testing-library/user-event';

afterEach(() => {
  return cleanup();
});

describe('TodoItem components', () => {
  it('matches snapshot', () => {
    const todo: Todo = { title: 'test title', description: 'test description' };
    const { asFragment } = render(<TodoItem todo={todo} />, {});
    expect(asFragment()).toMatchSnapshot(); //現在のCSSが適用されたビジュアルを保存しテストのたびに確認する
  });
  it(' render todo', () => {
    const todo: Todo = { title: 'test title', description: 'test description' };
    render(<TodoItem todo={todo} />);
    expect(screen.getByText('test title')).toBeInTheDocument(); //titleのチェック
    expect(screen.getByText('test description')).toBeInTheDocument(); //descriptionのチェック
  });
  it('click todo', () => {
    const mockFunction = jest.fn();
    const todo: Todo = {
      title: 'test title',
      description: 'test description',
    };
    render(<TodoItem todo={todo} onclick={mockFunction} />);
    const todoItem = screen.getByText('test title')
      .parentElement as HTMLElement;
    userEvent.click(todoItem);
    expect(mockFunction).toHaveBeenCalledTimes(1); //mockFunctionが1回よびだされているか
  });
});
