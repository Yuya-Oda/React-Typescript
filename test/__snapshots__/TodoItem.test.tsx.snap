// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem components matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todoItem"
  >
    <div
      class="todoItemTitle"
    >
      test title
    </div>
    <div
      class="todoDescriptionInput"
    >
      test description
    </div>
  </div>
</DocumentFragment>
`;
